### PR TITLE
espressif: Add support for hardware Secure Boot

### PR DESCRIPTION
**Pull request by gustavonihei**
This PR intends to enable hardware-supported Secure Boot for currently supported Espressif chips:
- **ESP32 (revision 3)**
- **ESP32-S2**
- **ESP32-C3 (revision 3)**

This feature is a complement to MCUboot's verification and it's based on Espressif's Secure Boot V2:
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/security/secure-boot-v2.html

Once Secure Boot V2 is enabled, the ROM bootloader verifies the signature of the MCUboot bootloader using the RSA-3072 algorithm, achieving a complete Secure Boot solution.

### CODE REVIEW

File: boot/espressif/hal/src/secure_boot.c [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    /* Initialize all efuse block entries to invalid (max) value */
+    esp_efuse_block_t blocks[SECURE_BOOT_NUM_BLOCKS] = {[0 ... SECURE_BOOT_NUM_BLOCKS-1] = EFUSE_BLK_KEY_MAX};
+    /* Check if secure boot digests are present */
+    bool has_secure_boot_digest = false;
+    for (unsigned i = 0; i < SECURE_BOOT_NUM_BLOCKS; i++) {
+        bool tmp_has_key = esp_efuse_find_purpose(secure_boot_key_purpose[i], &blocks[i]);
+        if (tmp_has_key) { // For ESP32: esp_efuse_find_purpose() always returns True, need to check whether the key block is used or not.
+            tmp_has_key &= !esp_efuse_key_block_unused(blocks[i]);
+        }
+        has_secure_boot_digest |= tmp_has_key;
```
Comment (utzig) [inline boot/espressif/hal/src/secure_boot.c:168]: I would use logic operators instead of bitwise for `bool`, but I guess this also works.

**Review by utzig — COMMENTED** [review]
LGTM

**Comment by almir-okato** [comment]
> /cc @almir-okato

LGTM

**Pull request merged**